connector-webhook/src/main/java/com/redhat/cloud/notifications/connector/webhook/WebhookRouteBuilder.java

```diff
@@ -0,0 +1,75 @@
+package com.redhat.cloud.notifications.connector.webhook;
+
+import com.redhat.cloud.notifications.connector.ConnectorConfig;
+import com.redhat.cloud.notifications.connector.EngineToConnectorRouteBuilder;
+import com.redhat.cloud.notifications.connector.webhook.authentication.BasicAuthenticationProcessor;
+import com.redhat.cloud.notifications.connector.webhook.authentication.BearerTokenAuthenticationProcessor;
+import com.redhat.cloud.notifications.connector.webhook.authentication.InsightsTokenAuthenticationProcessor;
+import org.apache.camel.builder.endpoint.dsl.HttpEndpointBuilderFactory;
+import org.apache.camel.component.http.HttpComponent;
+import org.apache.http.conn.ssl.NoopHostnameVerifier;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import static com.redhat.cloud.notifications.connector.ConnectorToEngineRouteBuilder.SUCCESS;
+import static com.redhat.cloud.notifications.connector.ExchangeProperty.*;
+import static com.redhat.cloud.notifications.connector.webhook.ExchangeProperty.BASIC_AUTH_USERNAME;
+import static com.redhat.cloud.notifications.connector.webhook.ExchangeProperty.BEARER_TOKEN;
+import static com.redhat.cloud.notifications.connector.webhook.ExchangeProperty.INSIGHT_TOKEN_HEADER;
+import static com.redhat.cloud.notifications.connector.webhook.ExchangeProperty.TARGET_URL_NO_SCHEME;
+import static com.redhat.cloud.notifications.connector.webhook.ExchangeProperty.TRUST_ALL;
+import static com.redhat.cloud.notifications.connector.webhook.SslTrustAllManager.getSslContextParameters;
+import static org.apache.camel.LoggingLevel.INFO;
+
+
+@ApplicationScoped
+public class WebhookRouteBuilder extends EngineToConnectorRouteBuilder {
+
+    @Inject
+    ConnectorConfig webhookConnectorConfig;
+
+    @Override
+    public void configureRoute() {
+
+        configureHttpsComponent();
+
+        from(direct(ENGINE_TO_CONNECTOR))
+            .routeId(webhookConnectorConfig.getConnectorName())
+            .choice()
+                .when(exchangeProperty(INSIGHT_TOKEN_HEADER))
```

Comment:
Is `exchangeProperty(INSIGHT_TOKEN_HEADER)` a valid Camel predicate? What's the result if the value is null, empty, not null?

Reply:
Yes it is. I validated it on `InsightsTokenAuthenticationTest`